website/content/en/docs/reference/configuration/sources/pipe.md

```diff
@@ -0,0 +1,14 @@
+---
+title: Pipe
+description: Collect logs sent via [pipe](https://en.wikipedia.org/wiki/File_descriptor)
```

Comment:
This, right here, makes me a little uncomfortable with the component name `pipe`. The component reads from a numbered file descriptor, which may be a pipe but could also be something else. I grant that the most common use case is a pipe setup, and calling it `file_descriptor` is probably worse, so `pipe` may be the least bad choice, but I'm not particularly happy with it neither.

Reply:
Same here,  this would theoretically also work with other file descriptors like sockets/files/etc.  Not sure what the right play is.

Reply:
It seems like our options are to either

- leave implementation as is, and decide whether or not to mention this can work with other file descriptor types
- re-name it file_descriptor
- try to find a way in the implementation, to discover what type of file descriptor was provided and reject if it is not a pipe

Reply:
Renaming to `file_descriptor` makes sense to me.  It was something I considered as @mcasper and I were "pairing" to make these changes, but, I didn't want to add more work at the time.  Let us know if that's acceptable and we'll make it happen.

Reply:
`file_descriptor` also makes sense to me with the current implementation, although it's not necessarily the phrase that I personally would be looking for if I had our original issue that spurred this on.

Either way I don't have a better idea, so if there are no objections then I'll proceed with switching it over to `file_descriptor`

Reply:
@bruceg , do you have further thoughts on this one?

Reply:
We have discussed this a bit and came to the consensus that `file_descriptor` is the best option for this.